### PR TITLE
Allow to override PluginManager

### PR DESCRIPTION
**Pull request by kokosing**
Allow to override PluginManager

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/server/ServerMainModule.java [hunk 1]
```diff
@@ -434,7 +435,8 @@ protected void setup(Binder binder)
         jaxrsBinder(binder).bind(StatusResource.class);
 
         // plugin manager
-        binder.bind(PluginManager.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder, PluginLoader.class).setDefault()
+                .to(Key.get(PluginManager.class));
```
Comment (Praveen2112) [inline core/trino-main/src/main/java/io/trino/server/ServerMainModule.java:439]: Do we need a `Key` here ?

**Review by Praveen2112 — COMMENTED** [review]
There is a CI failure related to duplicate classes - Is it flaky or ?

**Comment by kokosing** [comment]
Addressed comments, please take a look

**Comment by kokosing** [comment]
Addressed comments, please take a look.

**Review by wendigo — APPROVED** [review]

**Review by Praveen2112 — APPROVED** [review]

**Pull request merged**